ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdStatistics.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015 Avanza Bank AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.avanza.ymer;
+
+import java.util.Set;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.openspaces.core.util.ConcurrentHashSet;
+
+class PersistedInstanceIdStatistics implements PersistedInstanceIdStatisticsMBean {
+
+	private final Set<Integer> readyForNumberOfPartitions = new ConcurrentHashSet<>();
+	private LongAdder calculationProgress;
+	private Long totalToCalculate;
+
+	public void resetStatisticsForJobExecution(Set<Integer> calculatingForPartitions) {
+		calculationProgress = new LongAdder();
+		totalToCalculate = 0L;
+		readyForNumberOfPartitions.removeIf((numberOfPartitions -> !calculatingForPartitions.contains(numberOfPartitions)));
+	}
+
+	public void addToCalculationProgress(int calculationProgress) {
+		this.calculationProgress.add(calculationProgress);
+	}
+
+	public void setTotalToCalculate(Long totalToCalculate) {
+		this.totalToCalculate = totalToCalculate;
+	}
+
+	public void addReadyForNumberOfPartitions(int numberOfPartitions) {
+		readyForNumberOfPartitions.add(numberOfPartitions);
+	}
+
+	@Override
+	public int[] getReadyForNumberOfPartitions() {
+		return readyForNumberOfPartitions.stream().mapToInt(x -> x).sorted().toArray();
+	}
+
+	@Override
+	public Long getCalculationProgress() {
```

Comment:
What is the use-case of having this number compared to the logging being done in `PersistedInstanceIdCalculationService`? When does use either?
If it is for answering the question "is everything done?" then maybe it would be better to have a getter that returns `boolean` instead?

Reply:
Yes, the point is mainly to have a way of answering the question if everything is done. I used progress so that you get an idea of how long the calculation process will be (and could be plotted if needed, but not sure if there is a use case for that). So yes, choosing a boolean instead would maybe be an option for this.

Reply:
I updated this to simply use a `boolean` instead now. As you say, if the need is to check the current progression, the logging from this service can be used instead.

Reply:
👍